### DISCUSSION TITLE
Watara Supervision adapter ?

### DISCUSSION THREAD

**Discussion by sylverb** [sanni/cartreader#671] [Hardware]
Hello,
it seems that there is support for Watara Supervision cartridges but I can't find any info on the adapter needed to read Watara Supervision carts. I don't really want to read some Watara Supervision carts but some "Hartung Game Master" (was called "Systema 2000" in the UK) carts.
Carts connector is the same than for the Watara Supervision so it could be an adapter for both Supervision and Game Master carts ! I have info on how to dump Game Master games (I built and adapter with lots of wire and chip reader/programmer to previously dump 2 Game Master undumped games), and I think it would be great to add Game Master support as there are still few games that are supposed to exist and that are not dumped yet. I just need info on how to build the Supervision adapter, if someone can give me some info about this adapter, it would be great ...

**Comment by Cephiros** [comment]
I just drew this up quickly, It should work if everything is correct, I am unable to test due to having no cartridges for this console. It wouldn't hurt to double check the pinouts, I believe I flipped them correctly from cartridge to console slot.

These were created under the assumption that the Supervision uses a (nearly) standard type edge connector at a pitch of 2.5 vs 2.54 that most use. If the footprint is way off, you may need to create a new footprint and symbol and recreate the PCB.

[WSV2SNS-gerber.zip](https://github.com/sanni/cartreader/files/10338543/WSV2SNS-gerber.zip)- Gerber Files
[Console Symbols.zip](https://github.com/sanni/cartreader/files/10338544/Console.Symbols.zip)- Symbols to Load schematic
[WSV2SNS.zip](https://github.com/sanni/cartreader/files/10338545/WSV2SNS.zip)- Schematic and PCB design files

**Reply by sylverb** [reply]
Thanks for the files, supervision is using 2.54mm (not 100% sure) pitch from what I've found ! I'll update if needed so it can be compatible with Game Master pinout !

**Reply by sylverb** [reply]
Ok I did check about Game Master pinout and it looks like VCC/GND are not at the same position, I don't know if it's possible to create an adaptor that could be compatible with both system. Any advices ? 
Game Master connector pinout :
<img width="230" alt="GM_Connector" src="https://user-images.githubusercontent.com/255405/210744987-ca88d7d8-3fea-43d5-9650-e26ee3f09448.png">
Watara Supervision pinout :
<img width="275" alt="WSV_Connector" src="https://user-images.githubusercontent.com/255405/210745596-ca4d3464-2619-4c70-b177-118463a0a07a.png">

**Reply by Cephiros** [reply]
It would seem these are completely different units. If you would like to try the supervision dumping method on the gamemaster carts, just move the Ax and Dx pins to the correct pin. The problems might stem from the control pins and missing a15 (a16 is not used according to the code) I am not sure how WR and RD would map to CE OE and WE, L0-L4 are missing as well, but likewise unused in dumping.

**Reply by Cephiros** [reply]
Im assuming OE could be RD and WE is WR(output enable-read, and wright enable-write) there is no ce, chip enable on supervision, so there would have to be at minimum, a code change to pull that signal...., my guess is low, to enable the chip for use. You probably could in theory just ground the ce pin.

**Reply by sylverb** [reply]
Remapping all the data/address/chip enable/... signals is not a problem (as long as they are wired which Is not the case for some needed pins with the PCB design you did post). The blocking issues I see are that pin 1 is VCC (+5V) for the Supervision while pin 1 shall be GND for the Game Master which is quite incompatible, and pin 21 is /RD on the Supervision (which should be set to 0 for reading) while it's VCC for the Game Master. Maybe it could work if it's possible to use controlled pins as VCC and GND, but it has to output enough current to power the memory chip and absorb the current also (only if pins can be configured to act as GND and not as high impedance input).
I'm starting to think that a different adapter will be needed for the Game Master just because of that :( Unless a physical switch could make it possible to select between Supervision to Game Master power supply configuration.

**Reply by sylverb** [reply]
If it was not clear my idea was to have a single adapter for both Supervision and Game Master cartridges (as the cartridge port are the same on both systems), Don't know if it's something possible now and probably having 2 different adapters is the simplest solution.

**Reply by Cephiros** [reply]
I dont believe it would be possible, the mapping is completly different. Even if you did remap all the pins in code and pun a switch on the vcc/gnd pin to switch, the inverted pin is completely different (vcc is 21 on game master, rd on supervision, and gnd is 19 on supervision, which is nc on game master, so in theory you could just connect that to ground and be fine on both, leaving the pwr on a control pin problem.

**Reply by Cephiros** [reply]
Those connectors are cheap, if they are 2.54 pitch. Might be easier to make an adaptor with both (2 slots) on it, if you have both systems.

**Reply by PsyK0p4T** [reply]
Hi @sylverb, take a look at your no-intro inbox (if not done already).

Having two dedicated adapters is not a huge problem here. I mean, not so many people will want to dump one of these systems. And the two will be even rarer ^^ But if it's possible and easy to do so, then sure.

**Reply by sylverb** [reply]
Probably having 2 adaptors or one with 2 slots is the way to go ! I'll try to design one for the Game Master (kicad is not what I'm using often) but I'll check first if pitch is 2.5 or 2.54 (the connector I'm using was removed from a dead Supervision so I'm not sure, all I know is that Supervision cart port can be used to dump Game Master cartridges).
@PsyK0p4T : I already answered you ;)

**Reply by sylverb** [reply]
@Cephiros could you please share your console footprint library ? It looks like I need it to recreate a new PCB (maybe I could succeed without that but with the only way I know I'd need the footprint ;)

**Reply by Cephiros** [reply]
[footprints.zip](https://github.com/sanni/cartreader/files/10352909/footprints.zip)
sure, thought that might be the case, just wasn't sure due to no easy exporting like the symbol libs. KICad is somewhat new to me, I used easyeda for the longest time, peppered with bits of altium here and there.
